lib/enigma.rb

```diff
@@ -1,2 +1,29 @@
+require_relative './encryption'
+require_relative './decryption'
+require_relative './shifts'
+
 class Enigma
-end 
+  attr_reader :shifts
+
+  def initialize
+    @encryption = Encryption.new
+    @decryption = Decryption.new
+    @shifts = Shifts.new
+  end
+
+  def encrypt(message, key = @shifts.padded_random_number, date = @shifts.todays_date)
+    {
+      :encryption => @encryption.encrypt_message(message, @shifts.shifts(key, date)),
```

Comment:
Implementation of the optional arguments in the shifts class for the shifts method! Tricky to figure this out, but so satisfying and cool to see it work!